internal/nodes/crd_translator.go

```diff
@@ -479,3 +489,29 @@ func BuildL7SSORule(host, key string, vsNode AviVsEvhSniModel) {
 	serviceMetadataObj.CRDStatus = crdStatus
 	vsNode.SetServiceMetadata(serviceMetadataObj)
 }
+
+func BuildL7Rule(host, key, l7RuleName, namespace string, vsNode AviVsEvhSniModel) {
+	deleteL7RuleCase := false
+	l7Rule, err := lib.AKOControlConfig().CRDInformers().L7RuleInformer.Lister().L7Rules(namespace).Get(l7RuleName)
+	if err != nil {
+		utils.AviLog.Debugf("key: %s, msg: No L7Rule found for virtualhost: %s msg: %v", key, host, err)
+		deleteL7RuleCase = true
+	} else if l7Rule.Status.Status == lib.StatusRejected {
+		// do not apply a rejected L7Rule, this way the VS would retain
+		return
+	}
+	generatedFields := vsNode.GetGeneratedFields()
+	if !deleteL7RuleCase {
+		utils.AviLog.Debugf("key: %s, msg: applying l7 Rule %s", key, l7Rule.Name)
+		copier.CopyWithOption(vsNode, &l7Rule.Spec, copier.Option{DeepCopy: true})
```

Comment:
This VSnode can be a sharedVS(Parent). With copy option, we are copying all fields to vsnode. How are we making sure that only fields applicable to parent VS is stored and rest is nullified?

Reply:
we are doing nullification for child vs properties which controller restricts . for Dedicated and parent vs all properties can be applied.